util/db/repository.go

```diff
@@ -363,29 +368,31 @@ func (db *db) updateCredentialsSecret(credsInfo *settings.RepositoryCredentials,
 	credsInfo.TLSClientCertDataSecret = setSecretData(credSecretPrefix, r.Repo, secretsData, credsInfo.TLSClientCertDataSecret, r.TLSClientCertData, tlsClientCertData)
 	credsInfo.TLSClientCertKeySecret = setSecretData(credSecretPrefix, r.Repo, secretsData, credsInfo.TLSClientCertKeySecret, r.TLSClientCertKey, tlsClientCertKey)
 	for k, v := range secretsData {
-		err := db.upsertSecret(k, v)
+		_, err := db.upsertSecret(k, v, UpsertOptions{})
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (db *db) updateRepositorySecrets(repoInfo *settings.Repository, r *appsv1.Repository) error {
+func (db *db) UpdateRepositorySecrets(repoInfo *settings.Repository, r *appsv1.Repository, upsertOpts UpsertOptions) ([]*apiv1.Secret, error) {
```

Comment:
`argocd-util config repo-create` should not require connecting to the k8s cluster. I understand you want to reuse  ArgoDB code. Instead of using `UpsertOptions` I suggest to use `fake.NewSimpleClientset` initialized with default `argocd-cm` ConfigMap and `argocd-secret` Secret and print difference introduced by ArgoDB. 

https://github.com/argoproj/argo-cd/blob/991894e30535f6289fb76efa7a4f355e04a35dfa/cmd/argocd-util/commands/settings.go#L123